static/js/components/Periodogram.jsx

```diff
@@ -0,0 +1,573 @@
+                            {instruments.map((instrument) => (
+                              <MenuItem key={instrument} value={instrument}>
+                                {instrument}
+                              </MenuItem>
+                            ))}
+                          </Controller>
+                        </FormControl>
+                      </Grid>
+                      <Grid item xs={12}>
+                        <FormControl>
+                          <InputLabel
+                            className={classes.items}
+                            id="FilterSourceSelect"
+                          >
+                            Filter
+                          </InputLabel>
+                          <p />
+                          <Controller
+                            as={Select}
+                            labelid="FilterSourceSelectLabel"
+                            name="filter"
+                            control={control}
+                            defaultValue={params.filter}
+                            className={classes.items}
+                          >
+                            {filters.map((filt) => (
+                              <MenuItem key={filt} value={filt}>
+                                {filt}
+                              </MenuItem>
+                            ))}
+                          </Controller>
+                        </FormControl>
+                      </Grid>
+                      <Grid item xs={12}>
+                        <FormControl>
+                          <InputLabel className={classes.items} id="OFACSelect">
+                            ofac
+                          </InputLabel>
+                          <Controller
+                            as={<Input type="number" rules={rules} />}
+                            name="ofac"
+                            control={control}
+                            defaultValue={params.ofac}
+                            className={classes.items}
+                          />
+                        </FormControl>
+                      </Grid>
+                      <p />
+                      <Grid item xs={8}>
+                        {bestp && (
+                          <>
+                            <Typography>
+                              <b>Period</b> = {bestp?.toFixed(8)} d
+                            </Typography>
+                            <CopyToClipboard
+                              text={bestp}
+                              onCopy={() =>
+                                dispatch(
+                                  showNotification(
+                                    `Copied period (${bestp?.toFixed(
+                                      8
+                                    )} d) to clipboard.`
+                                  )
+                                )
+                              }
+                            >
+                              <Button
```

Comment:
The button text wraps over three lines, so could use tweaking

Reply:
Done.